fili-core/src/main/java/com/yahoo/bard/webservice/web/responseprocessors/EtagCacheResponseProcessor.java

```diff
@@ -0,0 +1,146 @@
+            QuerySigningService<Long> querySigningService,
+            ObjectMapper mapper
+    ) {
+        this.next = next;
+        this.dataCache = dataCache;
+        this.querySigningService = querySigningService;
+        this.mapper = mapper;
+        this.writer = mapper.writer();
+        this.respnoseCache = null;
+    }
+
+    @Override
+    public ResponseContext getResponseContext() {
+        return next.getResponseContext();
+    }
+
+    @Override
+    public FailureCallback getFailureCallback(DruidAggregationQuery<?> druidQuery) {
+        return next.getFailureCallback(druidQuery);
+    }
+
+    @Override
+    public HttpErrorCallback getErrorCallback(DruidAggregationQuery<?> druidQuery) {
+        return next.getErrorCallback(druidQuery);
+    }
+
+    @Override
+    public void processResponse(JsonNode json, DruidAggregationQuery<?> druidQuery, LoggingContext metadata) {
+        // make sure JSON response comes with status code
+        if (!json.has(DruidJsonResponseContentKeys.STATUS_CODE.getName())) {
+            logAndGetErrorCallback(ErrorMessageFormat.STATUS_CODE_MISSING_FROM_RESPONSE.format(), druidQuery);
+            return;
+        }
+
+        int statusCode = json.get(DruidJsonResponseContentKeys.STATUS_CODE.getName()).asInt();
+        if (statusCode == Status.NOT_MODIFIED.getStatusCode()) {
+            try {
+                ((ObjectNode) json).set(
+                        DruidJsonResponseContentKeys.RESPONSE.getName(),
+                        respnoseCache == null
+                                ? mapper.readTree(
+                                        dataCache.getDataValue(DruidJsonResponseContentKeys.RESPONSE.getName())
+                        )
+                                : respnoseCache
+                );
+            } catch (IOException ioe) {
+                throw new IllegalStateException(ioe);
+            }
+        } else if (statusCode == Status.OK.getStatusCode()) {
+            // make sure JSON response comes with etag
+            if (!json.has(DruidJsonResponseContentKeys.ETAG.getName())) {
+                logAndGetErrorCallback(ErrorMessageFormat.ETAG_MISSING_FROM_RESPONSE.format(), druidQuery);
+                return;
+            }
+
+            try {
+                dataCache.set(
+                        DruidJsonResponseContentKeys.RESPONSE.getName(),
+                        querySigningService.getSegmentSetId(druidQuery).orElse(null),
+                        writer.writeValueAsString(json.get(DruidJsonResponseContentKeys.RESPONSE.getName()))
+                );
+                dataCache.set(
+                        DruidJsonResponseContentKeys.ETAG.getName(),
+                        querySigningService.getSegmentSetId(druidQuery).orElse(null),
+                        writer.writeValueAsString(json.get(DruidJsonResponseContentKeys.ETAG.getName()))
+                );
+            } catch (JsonProcessingException exception) {
```

Comment:
My point was that it's not used in the body of the `catch` block. If we're intentionally ignoring a caught exception, it should be labeled as such (ie. called `ignored` or similar), otherwise we should use the caught exception in some way (often logging and wrapping in a more useful exception)